Ack ElasticSearch 5.0 changes regarding heap size

nubunto opened the pull request:
The heap size in ElasticSearch 5.0 now comes from the env var
"ES_JAVA_OPTS", as opposed to the old "JAVA_OPTS". This change strives
for better compatibility with other ElasticSearch versions, and acks
ES_JAVA_OPTS as one of the options of setting the heap size.

nubunto committed:
Ack ElasticSearch 5.0 changes regarding heap size

The heap size in ElasticSearch 5.0 now come from the env var
"ES_JAVA_OPTS", as opposed to the old "JAVA_OPTS". This change strives
for better compatibility with other ElasticSearch versions, and acks
ES_JAVA_OPTS as one of the options of setting the heap size.

gbjk commented on scheduler/src/main/java/org/apache/mesos/elasticsearch/scheduler/configuration/ExecutorEnvironmentalVariables.java:
```diff
@@ -67,6 +68,7 @@ private void populateEnvMap(Configuration configuration) {
         if (configuration.isFrameworkUseDocker()) {
             addToList(native_mesos_library_key, native_mesos_library_path);
         }
+        addToList(ES_JAVA_OPTS, getHeapSpaceString(configuration));
```
Errr, no, sorry, this is really wrong.

ES_JAVA_OPTS="192m" is blatantly wrong.

You should be aiming for: "-Xms192m -Xmx192m".
Treat the space string as part of what you set

nubunto committed:
Correctly handle ES_JAVA_OPTS

nubunto commented:
@gbjk view changes, please.

nubunto committed:
Further changes to make ES 5.0 work

- Removed the ES_HEAP_SIZE env var *entirely*, since ES 5.0 shuts it all
down if it finds it.
- Removed the args that are passed to ES, since he doesn't appear to
accept them anymore. Apparently they have to be given via
elasticsearch.yml.
- Updated the way ES_JAVA_OPTS environment variable is set, as opposed
to ES_HEAP_SIZE. The format is -Xms[MEM][m|g] -Xmx[MEM][m|g].

gbjk commented:
@nubunto - I think this is actually wrong, right? Wasn't it superceded ?

nubunto commented:
@gbjk yes, I need to push the changes I've made last week.

nubunto commented:
Wait, no, I have the changes that made it work in-place already with the latest commit.
The only piece missing is the max memory map area increase. Why would this be wrong?

gbjk commented:
@nubunto - I'm sure I had something in my head at the time, but ... now it's gone.

As it stands, this looks right.
